atomic_reactor/plugins/post_push_operator_manifest.py

```diff
@@ -0,0 +1,176 @@
+    get_omps_config,
+    get_koji_path_info,
+    NO_FALLBACK,
+)
+from atomic_reactor.util import (
+    get_platforms,
+    is_isolated_build,
+    is_scratch_build,
+    has_operator_manifest,
+    get_retrying_requests_session,
+)
+
+
+class PushOperatorManifestsPlugin(PostBuildPlugin):
+    """
+    Push operator manifest to appregistry via OMPS service
+    """
+    key = PLUGIN_PUSH_OPERATOR_MANIFESTS_KEY
+    is_allowed_to_fail = False
+
+    DOWNLOAD_DIR = 'operators_artifacts'
+
+    def is_orchestrator(self):
+        """
+        Check if the plugin is running in orchestrator.
+
+        :return: bool
+        """
+        if get_platforms(self.workflow):
+            return True
+        return False
+
+    def should_run(self):
+        """
+        Check if the plugin should run or skip execution.
+
+        :return: bool, False if plugin should skip execution
+        """
+        if not self.is_orchestrator():
+            self.log.warning("%s plugin set to run on worker. Skipping", self.key)
+            return False
+
+        if not get_omps_config(self.workflow, None):
+            self.log.info("Integration with OMPS is not configured. Skipping")
+            return False
+
+        if not has_operator_manifest(self.workflow):
+            self.log.info("Not an operator build. Skipping")
+            return False
+
+        if is_scratch_build():
+            self.log.info('Scratch build. Skipping')
+            return False
+
+        if is_rebuild(self.workflow):
+            self.log.info('Autorebuild. Skipping')
+            return False
+
+        if is_isolated_build():
+            self.log.info('Isolated build. Skipping')
+            return False
+
+        return True
+
+    def get_koji_operator_manifest_url(self):
+        """Construct URL for downloading manifest from koji task work server_dir
+
+        :rtype: str
+        :return: URL of koji operator manifests archive
+        """
+
+        server_dir = get_koji_upload_dir(self.workflow)
+
+        pathinfo = get_koji_path_info(self.workflow, NO_FALLBACK)
+        file_url = "{}/work/{}/{}".format(
+            pathinfo.topdir, server_dir, OPERATOR_MANIFESTS_ARCHIVE)
+
+        return file_url
+
+    def download_file(self, url, dest_filename):
+        """Downloads file specified by URL
+
+        :param url: file url
+        :param dest_filename: filename to be used for downloaded content
+
+        :return: file path of downloaded content
+        """
+        self.log.debug('Downloading file: %s', url)
+
+        workdir = self.workflow.source.get_build_file_path()[1]
+        dest_dir = os.path.join(workdir, self.DOWNLOAD_DIR)
+        dest_path = os.path.join(dest_dir, dest_filename)
+        if not os.path.exists(dest_dir):
+            os.makedirs(dest_dir)
+
+        req_session = get_retrying_requests_session()
+        request = req_session.get(url, stream=True,
+                                  verify=False)  # TODO configurable
```

Comment:
Just a note to check this TODO before final review

Reply:
I made it configurable